lib/taurus/core/util/fqdn.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+#############################################################################
+##
+# This file is part of Taurus
+##
+# http://taurus-scada.org
+##
+# Copyright 2011 CELLS / ALBA Synchrotron, Bellaterra, Spain
+##
+# Taurus is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+##
+# Taurus is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+##
+# You should have received a copy of the GNU Lesser General Public License
+# along with Taurus.  If not, see <http://www.gnu.org/licenses/>.
+##
+#############################################################################
+"""
+This module contains provides the fqdn_no_alias function.
+"""
+
+import socket
+from taurus import warning
+
+
+def fqdn_no_alias(hostname):
+    """
+    Normalize hostname so that it is a full name (including domain name).
+
+    .. note:: this is similar to socket.getfqdn, but it avoids translating
+              aliases into their "real" address.
+    """
+    if hostname == "localhost" or "." in hostname:
```

Comment:
I don't see to much benefice of this check. Moreover,  the hostname validation is skipped for these cases.

Reply:
The benefit is to avoiding/minimizing network connections during name validations in the most common cases (DB in localhost or "well configured" systems where the TANGO_HOST info is configured with the full domain info). I think it is worth keeping it.

My only concern is whether `"." in hostname` is the right check, but AFAIK it is good enought for now.